base/ca/src/main/java/com/netscape/cmscore/dbs/CertificateRepository.java

```diff
@@ -1180,13 +1180,10 @@ public Enumeration<CertRecord> findCertRecords(String filter)
             throws EBaseException {
         Enumeration<CertRecord> e = null;
 
-        try (DBSSession s = dbSubsystem.createSession()) {
-            //e = s.search(getDN(), filter);
-            CertRecordList list = findCertRecordsInList(filter, null, "serialno", 10);
-            int size = list.getSize();
+        CertRecordList list = findCertRecordsInList(filter, null, "serialno", 10);
+        int size = list.getSize();
 
-            e = list.getCertRecords(0, size - 1);
-        }
+        e = list.getCertRecords(0, size - 1);
```

Comment:
Hmm.. there are two `DBSSession` objects here, the one created in `findCertRecords()` and the one created in `findCertRecordsInList()`. The `CertRecordList` is created with the second `DBSSession` which also provides a connection to LDAP.

IIUC the `CertRecordList.getCertRecords()` will only work if the LDAP connection is still active, but at this point the second `DBSSession` is already closed (which is supposed to release the LDAP connection), so I'm suspecting that the code is currently working because we still have the first `DBSSession` even though we don't use it directly, so if we remove the first `DBSSession` it might introduce problems in some scenarios that we don't test in CI.

I'm currently working to change the default `minConns` to `0`. Let me test this PR against my changes, if there's no problem it's probably safe to merge.

Reply:
It would be very strange if the `CertRecordList` is valid because of an existing session which is unrelated with the query performed. It could be even in a separate thread. It would require more investigation to understand the connection.
However, if there is an extra tests to verify this it is better.

Reply:
I'm not too familiar with how ldapjdk or the connection factory works, but IIUC it involves cloning the `LDAPConnection` object, so maybe somehow they are still related. Regardless, I'm going to test this anyway. I'll let you know.